packages/@aws-cdk/aws-cloudfront/lib/experimental/edge-function.ts

```diff
@@ -252,11 +252,44 @@ function addEdgeLambdaToRoleTrustStatement(role: iam.IRole) {
 // This should be DRY'ed up (exported by @aws-lambda) before this is marked as stable.
 function calculateFunctionHash(fn: lambda.Function) {
   const stack = Stack.of(fn);
+
   const functionResource = fn.node.defaultChild as CfnResource;
+
   // render the cloudformation resource from this function
   const config = stack.resolve((functionResource as any)._toCloudFormation());
+  // config is of the shape: { Resources: { LogicalId: { Type: 'Function', Properties: { ... } }}}
```

Comment:
Why is this needed all of the sudden?

Reply:
Because there were new properties added to the `CfnFunction` class, and that changes the order of the keys in `config`. Without sorting them, we will be replacing user's Lambda versions when they upgrade.